src/object/srv_obj_migrate.c

```diff
@@ -634,21 +633,7 @@ mrone_obj_fetch(struct migrate_one *mrone, daos_handle_t oh, d_sg_list_t *sgls,
 	if (daos_oclass_grp_size(&mrone->mo_oca) > 1)
 		flags |= DIOF_TO_LEADER;
 
-	/**
-	 * For EC data migration, let's force it to do degraded fetch,
-	 * make sure reintegration will not fetch from the original
-	 * shard, which might cause parity corruption.
-	 */
-	obj = obj_hdl2ptr(oh);
-	if (iods[0].iod_type != DAOS_IOD_SINGLE &&
-	    daos_oclass_is_ec(&mrone->mo_oca) &&
-	    is_ec_data_shard(obj, mrone->mo_dkey_hash, mrone->mo_oid.id_shard) &&
-	    obj_ec_parity_alive(oh, mrone->mo_dkey_hash, NULL))
-		flags |= DIOF_FOR_FORCE_DEGRADE;
-
-	obj_decref(obj);
-
-	rc = dsc_obj_fetch(oh, mrone->mo_epoch, &mrone->mo_dkey,
```

Comment:
just confirm why don't need to force degraded fetch now?
is it because after this PR the "po_rebuilding = 1" for failed tgt is set no need not the FORCE_DEGRADED flag?

Reply:
oh, no. because we will use exact epoch to fetch the data, so it does not need to reconstruct the data from the parity, to make sure the epoch of the data shard will not ruin the data, (see above),  anymore.